src/seedu/addressbook/data/person/ReadOnlyPerson.java

```diff
@@ -82,4 +82,15 @@ default String getAsTextHidePrivate() {
         }
         return builder.toString();
     }
+
+    /**
+     * Returns a concatenated version of the printable strings of each object.
+     */
+    default String getPrintableString(Printable ... printables){
+        String retString = "";
+        for (Printable p : printables){
```

Comment:
Try to use a StringJoiner for string concatenation in a loop for readability and flexibility.